src/components/abstractComponent.ts

```diff
@@ -0,0 +1,579 @@
+   * Checks if the Component has a fixed height or if it grows to fill available space.
+   * Returns false by default on the base Component class.
+   */
+  fixedHeight(): boolean;
+  /**
+   * Detaches a Component from the DOM. The Component can be reused.
+   *
+   * This should only be used if you plan on reusing the calling Component. Otherwise, use destroy().
+   *
+   * @returns The calling Component.
+   */
+  detach(): this;
+  /**
+   * Adds a callback to be called when the Component is detach()-ed.
+   *
+   * @param {GenericComponentCallback} callback
+   * @return {IComponent} The calling Component.
+   */
+  onDetach(callback: GenericComponentCallback<D>): this;
+  /**
+   * Removes a callback to be called when the Component is detach()-ed.
+   * The callback is identified by reference equality.
+   *
+   * @param {GenericComponentCallback} callback
+   * @return {IComponent} The calling Component.
+   */
+  offDetach(callback: GenericComponentCallback<D>): this;
+  /**
+   * Gets the parent ComponentContainer for this Component.
+   */
+  parent(): ComponentContainer;
+  /**
+   * Sets the parent ComponentContainer for this Component.
+   * An error will be thrown if the parent does not contain this Component.
+   * Adding a Component to a ComponentContainer should be done
+   * using the appropriate method on the ComponentContainer.
+   */
+  parent(parent: ComponentContainer): this;
+  parent(parent?: ComponentContainer): any;
+  /**
+   * @returns {Bounds} for the component in pixel space, where the topLeft
+   * represents the component's minimum x and y values and the bottomRight represents
+   * the component's maximum x and y values.
+   */
+  bounds(): Bounds;
+  /**
+   * Removes a Component from the DOM and disconnects all listeners.
+   */
+  destroy(): void;
+  /**
+   * Gets the width of the Component in pixels.
+   */
+  width(): number;
+  /**
+   * Gets the height of the Component in pixels.
+   */
+  height(): number;
+  /**
+   * Gets the origin of the Component relative to its parent.
+   *
+   * @return {Point}
+   */
+  origin(): Point;
+  /**
+   * Gets the origin of the Component relative to the root
+   *
+   * @return {Point}
+   */
+  originToRoot(): Point;
+  /**
+   * Gets the container holding the visual elements of the Component.
+   *
+   * Will return undefined if the Component has not been anchored.
+   *
+   * @return {IContent} content selection for the Component
+   */
+  content(): IContent;
+  /**
+   * Gets the top-level element of the component
+   */
+  element(): IContent;
+  /**
+   * Gets the root component of the hierarchy
+   *
+   * @return {IComponent}
+   */
+  root(): IComponent<any>;
+  /**
+   * Gets the translator for the component. A translator
+   * converts client coordinates to component coordinates and
+   * takes into account any transformations that have been applied
+   * to the element.
+   *
+   * @return {Utils.ITranslator}
+   */
+   translator(): Utils.Translator;
```

Comment:
Ugh this was me trying to come up with a way of associating the correct translator type with the component type. The goal was for a third party to be able to, given a generic IComponent, determine the offset of that component in screen space. Specifically the mouse dispatcher needs to be able to determine the offset of the component in order to dispatch interactions correctly 

https://github.com/palantir/plottable/pull/3209/files#diff-b5b8bc221836e19ab1bae0224ffd72c2R53 


But I think we talked about a solution for making the translator a one-size-fits-all top level `svg` that should render this method obsolete.